source/_posts/2015-03-23-artsy-technology-stack-2015.markdown

```diff
@@ -0,0 +1,56 @@
+
+<center><img src='/images/2015-03-23-artsy-technology-stack-2015/artsy.png'></center>
+
+These web applications talk to a private Ruby API built with [Grape](https://github.com/intridea/grape), that serves JSON. We also have a more modern and better designed [public HAL+JSON API](https://developers.artsy.net). For historical reasons, both are hosted side-by-side on top of a big Rails app that used to be our original monolith. The API service runs on [AWS OpsWorks](http://aws.amazon.com/opsworks) and retrieves data from several [MongoDB](http://www.mongodb.com) databases hosted with [Compose](https://www.compose.io). It also uses [Apache Solr](http://lucene.apache.org/solr), being replaced by [Elastic Search](https://www.elastic.co) and [Google Custom Search](https://www.google.com/cse). The API service also heavily relies on [Memcached](http://memcached.org).
+
+<center><img src='/images/2015-03-23-artsy-technology-stack-2015/developers.png'></center>
+
+Our partners upload artworks, artists and other metadata via a home-grown content-management system built entirely from scratch in 2014 on Ruby-on-Rails. This was a deliberate "boring" choice that continues to serve us very well. We have adopted a common model for admin-type apps with a shared component library and a reusable UI, all implemented as Rails engines. Using these components we are able to quickly and easily compose beautiful and useful applications - we have built dedicated systems to manage fairs and auctions. We standardized on related services as well - for example, our customers interact with us via [Intercom](https://www.intercom.io).
+
+<center><img src='/images/2015-03-23-artsy-technology-stack-2015/cms.png'></center>
+
+Our family of mobile applications includes [Artsy for iOS](http://iphone.artsy.net), which is a hybrid app written in Objective-C, and a [bidding kiosk](https://github.com/artsy/eidolon), written in Swift. Both are open-source [here](https://github.com/artsy/eigen) and [here](https://github.com/artsy/eidolon).
+
+<center><img src='/images/2015-03-23-artsy-technology-stack-2015/folio.jpg'></center>
+
+A lot of data, including the artwork Similarity graph that powers the Art Genome Project, is processed offline by a generic job engine, written in Ruby. We take data snapshots from MongoDB, run jobs on the data and export data back to the database. Other recently rewritten services include image processing, which creates thumbnails, image tiles for deep zoom and watermarks high quality JPEGs. Several new applications use [PosgreSQL](http://www.postgresql.org).
```

Comment:
Yeah, we always right TAGP as "The Art Genome Project."

Looking forward to this post. It really helps us non-tech people be able to
talk about the engineering team's awesome work in a more informed way. As
this is very fundamental to Artsy, you might consider sending this post to
the team at large (you may want to explain what a tech stack is).

- J

On Tue, Mar 24, 2015 at 11:58 AM, mzikherman <notifications@github.com>
wrote:

> In source/_posts/2015-03-23-artsy-technology-stack-2015.markdown
> <https://github.com/artsy/artsy.github.com/pull/78#discussion_r27042313>:
>
> > +
> > +<center><img src='/images/2015-03-23-artsy-technology-stack-2015/artsy.png'></center>
> > +
> > +These web applications talk to a private Ruby API built with [Grape](https://github.com/intridea/grape), that serves JSON. We also have a more modern and better designed [public HAL+JSON API](https://developers.artsy.net). For historical reasons, both are hosted side-by-side on top of a big Rails app that used to be our original monolith. The API service runs on [AWS OpsWorks](http://aws.amazon.com/opsworks) and retrieves data from several [MongoDB](http://www.mongodb.com) databases hosted with [Compose](https://www.compose.io). It also uses [Apache Solr](http://lucene.apache.org/solr), being replaced by [Elastic Search](https://www.elastic.co) and [Google Custom Search](https://www.google.com/cse). The API service also heavily relies on [Memcached](http://memcached.org).
> > +
> > +<center><img src='/images/2015-03-23-artsy-technology-stack-2015/developers.png'></center>
> > +
> > +Our partners upload artworks, artists and other metadata via a home-grown content-management system built entirely from scratch in 2014 on Ruby-on-Rails. This was a deliberate "boring" choice that continues to serve us very well. We have adopted a common model for admin-type apps with a shared component library and a reusable UI, all implemented as Rails engines. Using these components we are able to quickly and easily compose beautiful and useful applications - we have built dedicated systems to manage fairs and auctions. We standardized on related services as well - for example, our customers interact with us via [Intercom](https://www.intercom.io).
> > +
> > +<center><img src='/images/2015-03-23-artsy-technology-stack-2015/cms.png'></center>
> > +
> > +Our family of mobile applications includes [Artsy for iOS](http://iphone.artsy.net), which is a hybrid app written in Objective-C, and a [bidding kiosk](https://github.com/artsy/eidolon), written in Swift. Both are open-source [here](https://github.com/artsy/eigen) and [here](https://github.com/artsy/eidolon).
> > +
> > +<center><img src='/images/2015-03-23-artsy-technology-stack-2015/folio.jpg'></center>
> > +
> > +A lot of data, including the artwork Similarity graph that powers the Art Genome Project, is processed offline by a generic job engine, written in Ruby. We take data snapshots from MongoDB, run jobs on the data and export data back to the database. Other recently rewritten services include image processing, which creates thumbnails, image tiles for deep zoom and watermarks high quality JPEGs. Several new applications use [PosgreSQL](http://www.postgresql.org).
>
> 'the' should be capitalized in The Art Genome Project (I think).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/artsy/artsy.github.com/pull/78/files#r27042313>.
>



-- 

[A.] Jessica Backus

Artsy <http://Artsy.net>


jessica@art <jessica@art.sy>sy <http://symail.com>.net

+1 718 685 6543

                artsy.net/jessica

                Twitter <http://www.twitter.com/that_j>



Artsy Education <http://artsy.net/feature/artsy-education> | The Art Genome
Project <http://artsy.net/theartgenomeproject>


We're hiring! Artsy seeks Contributors to The Art Genome Project, Content
Partner Liaisons (NY, LA, Hong Kong), Editorial Associates, Engineers and
Interns <https://artsy.net/about/jobs>